lib/ecto/schema.ex

```diff
@@ -2041,12 +2041,18 @@ defmodule Ecto.Schema do
                            "(no time zone information) or :utc_datetime (time zone is set to UTC)"
   end
 
-  defp check_field_type!(name, type, opts) do
+  defp check_field_type!(name, type, opts, full_type \\ nil) do
+    full_type = if is_nil(full_type), do: type, else: full_type
+
     cond do
       type == :any and !opts[:virtual] ->
         raise ArgumentError, "only virtual fields can have type :any, " <>
                              "invalid type for field #{inspect name}"
 
+      is_tuple(type) and tuple_size(type) == 2 and type |> elem(0) |> Ecto.Type.composite?() ->
+        inner_type = elem(type, 1)
+        check_field_type!(name, inner_type, opts, type)
```

Comment:
If it is a two element tuple, it is always composite:

```suggestion
      is_tuple(type) and tuple_size(type) == 2 ->
        check_field_type!(name, elem(type, 1), opts, type)
```

We will also need a three element tuple check for parameterized types. :D

Reply:
@josevalim 
> If it is a two element tuple, it is always composite:

Sorry, I do not understand that. If so then why we have a `Ecto.Type.composite?` function and how would let's say `{:jsonb, :jsonb}` work?

> We will also need a three element tuple check for parameterized types. :D

Ah right, I forgot about that